### DISCUSSION TITLE
Custom services with custom resolver

### DISCUSSION THREAD

**Discussion by fabianboerner** [TriPSs/nestjs-query#171] [General]
Is it possible that the service in a custom resolved is incompatible with a custom crudservice?

I currently dont see how i can create a custom service and use it in the custom resolver-.

**Comment by TriPSs** [comment]
Could you eleborate a bit more? When using a custom crud service you can just use the `@inject(ServiceClass)` to inject it.

**Comment by fabianboerner** [comment]
I have this
```
import {IndeedDetailedCampaignStatEntryDto} from "./dtos/indeed-detailed-campaign-stat-entry.dto";
import {Args, ID, Query, Resolver} from "@nestjs/graphql";
import {CRUDResolver} from "@ptc-org/nestjs-query-graphql";
import {IndeedDetailedCampaignStatEntry} from "./schemas/indeed-detailed-campaign-stat-entry.schema";
import {InjectQueryService, QueryService} from "@ptc-org/nestjs-query-core";
import {IndeedDetailedCampaignStatsEntryService} from "./services/indeed-detailed-campaign-stats-entry.service";

@Resolver(() => IndeedDetailedCampaignStatEntryDto)
export class IndeedAnalyticsResolver extends CRUDResolver(IndeedDetailedCampaignStatEntryDto) {
  constructor(
    @InjectQueryService(IndeedDetailedCampaignStatEntry) readonly service: IndeedDetailedCampaignStatsEntryService
  ) {
    super(service);

  }

  @Query(() => [IndeedDetailedCampaignStatEntryDto], {nullable: true})
  async indeedDetailedCampaignStatEntryByPositionId(@Args({name: 'positionId', type: () => String}) positionId: string
  ) {
    return await this.indeedDetailedCampaignStatsEntryService.findGroupedByPositionId(positionId);
  }
}
```

and the custom service is incompatible with the custom resolvers servcice

**Reply by TriPSs** [reply]
Could it be a that `IndeedDetailedCampaignStatEntryDto` and `IndeedDetailedCampaignStatEntry` are not compatible? For example, if the DTO has `fieldOne: string` but the entity has `fieldOne?: string` they are not compatible.

**Reply by mareksuscak** [reply]
I'm also running into type issues here due to the fact that the type declarations are inconsistent - one requires a DTO, one requires a model and they're not exactly the same because some properties are internal and not meant to be exposed in the DTO.
